### ISSUE TITLE
Unable to create new "Terminal" session on Windows

### ISSUE THREAD

**Issue by datakurre** [robots-from-jupyter/robotlab#54]
Steps

1. install RobotLab on Windows
2. start RobotLab
3. File => New => Terminal

Expected

Terminal session opens

What actually happens

Terminal session opens and closes so fast that I'm unable to see what happens.

RobotLab launch terminal printed the following

```
[I 23:18:00.308 LabApp] EOF on FD 2688; stopping reading
[I 23:18:00.526 LabApp] Terminal 1 closed
TermSocket.open: 1
[I 23:18:00.526 LabApp] New terminal with specified name: 1
TermSocket.open: Opened 1
Exception in thread Thread-16:
Traceback (most recent call last):
  File "C:\Users\user\RobotLab\lib\threading.py", line 916, in _bootstrap_inner
    self.run()
  File "C:\Users\user\RobotLab\lib\threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\user\RobotLab\lib\site-packages\winpty\ptyprocess.py", line 331
, in _read_in_thread
    data = pty.read(4096)
  File "C:\Users\user\RobotLab\lib\site-packages\winpty\winpty_wrapper.py", line
 73, in read
    ReadFile(self.conout_pipe, data, length, None, None)
OSError: exception: access violation writing 0x0000000000000000

[I 23:18:00.761 LabApp] EOF on FD 2680; stopping reading
[I 23:18:00.980 LabApp] Terminal 1 closed
```

**Comment by datakurre** [comment]
I was also unable to create Robot Framework notebook, because of

```
Unhandled Exception: System.BadImageFormatException: Could not load file
or assembly 'clr.pyd' or one of its dependencies. This assembly is built by
a runtime newer than the currently loaded runtime and cannot be loaded.
File name: 'clr.pyd'
```

I might have outdated Windows on my virtual machine. Installing updates and trying again...

**Comment by bollwyvl** [comment]
Yeah sounds gross. Windows is weird.

We should add a terminal test, I suppose, to see if we can catch these. I
also may have done a poor job of packaging something relevant... Hard to
tell!

**Comment by datakurre** [comment]
The latter issue with Robot Kernel was just outdated .NET on my virtual machine snapshot. (It is required, because we pack WhiteLibrary ja RobotKernel tries to import it when it is available.)

**Comment by bollwyvl** [comment]
Dang. I thought we shipped that, too, by way of pythonnet and deps. I guess
there's some things that have to be in c:\windows 32\

I guess over in RK have an optional dependency helper or something? I
forgot where I saw a good one. Sympy has a ton of optional deps...

**Comment by datakurre** [comment]
I could use setuptools API to check the existence of dependencies instead of just trying to import them, but it would not help here, because it looks like dependencies are there and then importing pythonnet crashes the interpreter :)

**Comment by bollwyvl** [comment]
Wow!

So do we need to ship .net, too? Blearggh

**Comment by datakurre** [comment]
I believe it is enough to require up-to-date supported Windows version. The error means that something is built with newer versions of .NET than my old Windows 7 virtual machine save point had. I’ll make a new save point with all available packages and the next time I’ll see that, Windows 7 has been EOL’d long time ago.

**Comment by bollwyvl** [comment]
Thank goodness!

**Comment by datakurre** [comment]
This actual issue could be fixed by packaging the latest pywinpty == 0.5.5 with RobotLab. (The error is bug in the packaged 0.5.0 version.)

**Comment by bollwyvl** [comment]
ref https://github.com/spyder-ide/pywinpty/pull/109
